server/index.ts

```diff
@@ -0,0 +1,60 @@
+import dotenv from 'dotenv'
+import express from 'express'
+import mongoose from 'mongoose'
+/* eslint-disable import/first */
+dotenv.config()
+import { PORT, DB_URL } from './config/constants'
+import userRouter from './routes/user.route'
+import authRouter from './routes/auth.route'
+
+const app = express()
+const SERVER_PORT = PORT || 5000
+
+const formatResponse = (req, res, next) => {
+    res.formatResponse = (data, message, status?, errors?) => {
+        const newData = {}
+
+        Object.defineProperty(newData, 'data', {
+            value: data,
+            enumerable: true,
+        })
+
+        Object.defineProperty(newData, 'message', {
+            value: message,
+            enumerable: true,
+        })
+
+        if (errors) {
+            Object.defineProperty(newData, 'errors', {
+                value: errors,
+                enumerable: true,
+            })
+        }
+
+        if (status) {
+            res.status(status)
+        }
+
+        return res.json(newData)
+    }
+    next()
+}
+
+app.use(express.json())
```

Comment:
you have `express.json()` middleware and `return res.json(newData)` at the same time, do you need both of them? i guess you should use only one way

Reply:
Maybe I misunderstood something. The middleware express.json() needs to handle req with json bodies. My middleware converts res to json format.